src/game/Coordinate.java

```diff
@@ -3,16 +3,26 @@
 public class Coordinate {
 	private int x;
 	private int y;
-	private boolean valid = false;
+	private boolean valid;
 
 	public Coordinate(int x, int y) {
 		if (isValidValue(x) && isValidValue(y)) {
 			this.valid  = true;
 			this.x = x;
 			this.y = y;
+		} else {
+			this.valid = false;
+			this.x = -1;
```

Comment:
explicitly sets the x and y of an invalid coordinate to invalid values instead of 0